worlds/seaofthieves/Locations/Hunter/ProvisionsCooked/BurntAboard.py

```diff
@@ -0,0 +1,301 @@
+                name == HunterBurntAboard.L_H_CAT_BURN_ANCIENTSCALE or
+                name == HunterBurntAboard.L_H_CAT_BURN_PLENTIFIN or
+                name == HunterBurntAboard.L_H_CAT_BURN_WILDSPLASH or
+                name == HunterBurntAboard.L_H_CAT_BURN_DEVILFISH or
+                name == HunterBurntAboard.L_H_CAT_BURN_BATTLEGILL or
+                name == HunterBurntAboard.L_H_CAT_BURN_WRECKER or
+                name == HunterBurntAboard.L_H_CAT_BURN_STORMFISH):
+            return self.fishCategory
+
+        if name == HunterBurntAboard.L_H_CAT_BURN_LAND:
+            return self.landMeat
+
+        if name == HunterBurntAboard.L_H_CAT_BURN_BIG:
+            return self.bigFish
+
+        # default for it to be enabled
+        return 1
+
+
+class HunterBurntAboard(LocationsBase):
+    L_H_BURN = "Burn Anything (H)"
+
+    # cat names
+    L_H_CAT_BURN_SPLASHTAIL = "Burn Splashtail (H)"
+    L_H_CAT_BURN_PONDIE = "Burn Pondie (H)"
+    L_H_CAT_BURN_ISLEHOPPER = "Burn Islehopper (H)"
+    L_H_CAT_BURN_ANCIENTSCALE = "Burn Ancientscale (H)"
+    L_H_CAT_BURN_PLENTIFIN = "Burn Plentifin (H)"
+    L_H_CAT_BURN_WILDSPLASH = "Burn Wildsplash (H)"
+    L_H_CAT_BURN_DEVILFISH = "Burn Devilfish (H)"
+    L_H_CAT_BURN_BATTLEGILL = "Burn Battlegill (H)"
+    L_H_CAT_BURN_WRECKER = "Burn Wrecker (H)"
+    L_H_CAT_BURN_STORMFISH = "Burn Stormfish (H)"
+    L_H_CAT_BURN_LAND = "Burn Land Creature (H)"
+    L_H_CAT_BURN_BIG = "Burn Shark, Megalodon, or Kraken (H)"
+
+    def __init__(self, settings: SettingsHunterBurntAboard):
+        super().__init__()
+        self.x = [3, 0, 2]
+        self.settings = settings
+
+        self.add_any_sets()
+        self.add_fish_sets()
+        self.add_meat_set_land()
+        self.add_fish_set_big()
+
+    def add_any_sets(self):
+        do_rand: bool = self.settings.completeAny is not self.settings.Any.OFF
+        reg = RegionNameCollection()
+        reg.addFromList([Regions.R_SHIP_COOKER])
+        lgc = ItemReqEvalOr([ItemReqEvalAnd([Items.fishing_rod])])
+        wlc = WebLocationCollection([
+            WebLocation(WebItemJsonIdentifier(self.x[0], self.x[1], self.x[2]), reg, lgc)
+        ])
+        self.locations.append(LocDetails(self.L_H_BURN, wlc, do_rand))
+
+    def add_fish_sets(self):
+        self.add_fish_set_pondie()
+        self.add_fish_set_splashtail()
+        self.add_fish_set_islehopper()
+        self.add_fish_set_ancientscale()
+        self.add_fish_set_plentifin()
+        self.add_fish_set_wildsplash()
+        self.add_fish_set_devilfish()
+        self.add_fish_set_battlegill()
+        self.add_fish_set_wrecker()
+
+    def add_land_sets(self):
+        self.add_meat_set_land()
+
+    def addFishSetLoc(self, name: str, wlc: WebLocationCollection):
+        setting = int(self.settings.fishCategory)
+
+        do_rand: bool = setting == SettingsHunterBurntAboard.Fish.ON
+        # TODO not implemented
```

Comment:
Is this todo still valid?

Reply:
This is still TODO and it is a BUG.

Comment for programmer reference:
Most location sets have an option called ON. When ON is the chosen option (as apposed to ON_FOR_EACH or in fish case, CATEGORICAL_NAME/UNIQUE) we want a single location in the game that represents each "check" in the set. For example, if Burnt Fish Setting = ON I would expect the game to have a single ITEM that would be rewarded if I burn a fish. The randomizer needs to know that all these locations should not be submitted to the multiworld and they should not contain real items since we expect the player to not burn each fish. This is the purpose of the do_rand boolean, if set to false, we will not "add items and/or submit them to the multiworld". 

With the context above, we are missing a line of code that adds all the locations in the set as do_rand=false to the locations list. This was because it was complex to achieve.